native/acter/src/api/convo.rs

```diff
@@ -69,19 +69,16 @@ impl Convo {
                 // skip the state event
                 // if let Ok(AnySyncTimelineEvent::MessageLike(m)) = event.event.deserialize() {
                 if let Some(msg) = sync_event_to_message(&event.event, room.room_id().to_owned()) {
-                    self.set_latest_message(msg);
+                    self.set_latest_message(Box::new(msg));
                     return;
                 }
                 // }
             }
         }
     }
 
-    fn set_latest_message(&mut self, mut msg: RoomMessage) {
-        if let Some(mut event_item) = msg.event_item() {
-            msg.set_event_item(Some(event_item));
-        }
-        self.latest_message = Some(msg);
+    pub fn set_latest_message(&mut self, msg: Box<RoomMessage>) {
+        self.latest_message = Some(*msg);
```

Comment:
I am not convinced that piping the message through the fronted to then set it back on the backend is a good idea ...

but I as I am on a refactor for the entire section, it's fine for now...

Reply:
It's fine for now. I am currently looking at merging timeline and latest_room message in #915 .